Keen/Dataset/Datasets.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+
+namespace Keen.Core.Dataset
+{
+    using System.Threading.Tasks;
+    using ContractResolvers;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+    using Query;
+
+    public interface IDataset
```

Comment:
I would tend to prefer interfaces in their own file, and public API surface (types, fields, methods, etc.) should have doc comments.